### PR TITLE
DM-24642: Implement simple single-pass data pacer for alert stream simulator

### PR DESCRIPTION
**Pull request by spenczar**
## Description

This PR adds features to the rubin-alert-sim command to let a user replay an avro alert data file "at speed" - in other words, at the pace described by timestamps from the avro alerts themselves.

Early drafts of this code tried to do this on-the-fly: read in the data file one alert at a time, deserialize it, inspect the timestamp, and then reserialize and publish it into Kafka at the right moment. This turned out to be _far_ too slow. Avro serialization of alert packets takes approximately 2-3ms for HITS data, which meant that we could only send about 300 alerts per second; we need to send at more like 2,000 per second to really replicate Rubin's pace.

So, this PR takes a different approach. Work is done in two phases. In the first phase, a stream is "created": An alert file is read from disk and individual alerts are sent into Kafka with headers describing their time offset (in seconds) relative to the first alert in the file. In the second phase, we read from the first phase's Kafka stream, inspect _just the header_ to determine whether we need to sleep, and reinsert into Kafka.

These two phases are implemented as two subcommands. `rubin-alert-sim create-stream` does the first step, and `rubin-alert-sim play-stream` does the second.

Since these commands create topics on-the-fly, I found myself quickly getting mired in Kafka complexity. I also hit Kafka complexity when writing integration tests. To manage that complexity, I have written a small Kafka client wrapper, too.

#### Benchmark:

I am able to get about 600 alerts per second if I use a 10-partition destination topic with `play-stream`. This is about 200MB/s which is fairly quick. I can slightly increase the pace by running more brokers, but only at a severe cost in complexity; I think 600/s will have to suffice.

#### TODO:
 - [x] Documentation 
 - [x] Deduplicate test utility code
 - [x] Implement confluent's message serialization scheme for schema registry compatibility
 - [x] Command-line-based integration test
 - [x] Benchmark throughput

### CODE REVIEW

File: test/test_create.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        alert_file = testutils.mock_alert_file(alerts)
+
+        n_sent = creator.create("localhost:9092", topic_name, alert_file, 5.0, True)
+        self.assertEqual(n_sent, len(alerts))
+
+        kc = _kafka._KafkaClient("localhost:9092")
+        kc.subscribe(topic_name, 15)
+
+        msgs = kc.consumer.consume(n_sent, timeout=20)
+        self.assertAlmostEqual(len(msgs), n_sent)
```
Comment (morriscb) [inline test/test_create.py:49]: Why is this almost equal for an int?

File: test/test_play.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import astropy.time
+
+from streamsim import creator, player, _kafka
+from test import testutils
+
+
+@pytest.mark.integration_test
+class TestPlayIntegration(unittest.TestCase):
+    @staticmethod
+    def mock_alert(alert_id, timestamp):
```
Comment (morriscb) [inline test/test_play.py:33]: Do you want to use your utility code again here?

Comment (morriscb) [reply test/test_play.py:33]: Looking at it more, is this even used anymore?

Comment (spenczar) [reply test/test_play.py:33]: Oops, you're right. I added the utility code but forgot to remove this spot, which is where I first wrote it.

File: test/test_create.py (hunk 1 above)
Comment (spenczar) [reply test/test_create.py:49]: Right, this is a mistake.

**Comment by spenczar** [comment]
Agh, I wasn't thinking and didn't merge this the LSST Way, with a `--no-ff` merge into master. Oops.

**Pull request merged**